salt/netapi/rest_tornado/saltnado.py

```diff
@@ -921,14 +921,27 @@ def _disbatch_local(self, chunk):
         '''
         Dispatch local client commands
         '''
-        # Generate jid before triggering a job to subscribe all returns from minions
-        chunk['jid'] = salt.utils.jid.gen_jid()
+        # Generate jid and find all minions before triggering a job to subscribe all returns from minions
+        chunk['jid'] = salt.utils.jid.gen_jid() if not chunk.get('jid', None) else chunk['jid']
```

Comment:
i think you want gen_jid(self.opts)

Reply:
the function `def get_jid(opt=None)`  is exist on `develop` branch only.
on `2017.7`, `get_jid` doesn't has `opt` options.

Reply:
my bad, you're right :+1: